File: examples/sample_functions/ex.py
def sum(a: int, b: int) -> int:
    return a + b

def pow(a: int, b: int) -> int:
    return a ** b

def pow(a: int, b: int) -> int:
    result = 1
    for _ in range(b):
        result = result * a
    return result

def div(a: float, b: float) -> float:
    return a / b

def sum_array(v: list) -> int:
    result = 0
    for item in v:
        result += item
    return result

import sys
print(int(sys.argv[1]) + int(sys.argv[2]))

File: examples/sample_functions/ex.cpp
int sum(int a, int b)
{
    return a + b;
}

int pow(int a, int b)
{
    int result = 1;
    for (int i = 0; i < b; i++)
    {
        result *= a;
    }
    return result;
}

float div(float a, float b)
{
    return a / b;
}

int sum_array(std::vector<int> v)
{
    int result = 0;
    for (auto item : v)
    {
        result += item;
    }
    return result;
}

#include <string>
char* concat(char* a, char* b)
{
    std::string buffer = std::string(a) + std::string(b);
    char* output = new char[buffer.length() + 1];
    std::memcpy(output, buffer.c_str(), buffer.length());
    output[buffer.length()] = NULL;
    return output;
}